GameEngine/Characters/Player.cpp

```diff
@@ -29,33 +29,35 @@ Collider* Player::getCollider() {
 void Player::doAttack() {
 	// Create bullet and launch towards enemies
 	Bullet* bullet = new Bullet(new RECT{ position->left, position->top - 32, position->right, position->bottom - 32}, hdc);
-	std::thread bulletThread(&Bullet::move, bullet);
+	std::thread bulletThread(&Bullet::move, bullet); 
```

Comment:
I was thinking the threads originally because I wanted all the bullets to move as soon as they are shot. But I see how it makes sense to gather all the bullets and then move them each frame

Reply:
A follow up to this though, when might the investment for a thread be worth it? Or is it never? What if we had to shoot like, 10,000 bullets, or there were multiple players with multiple effects. 
How does League handle this?

Reply:
Have a look at https://github.com/id-Software/Quake/qw-qc/weapons.qc(440)
W_FireRocket or W_FireGrenade which is pretty similar to the bullets you are creating above. 
Nextthink is when the rocker/grenade is ticked again.

Reply:
https://en.wikipedia.org/wiki/Context_switch and https://en.wikibooks.org/wiki/C%2B%2B_Programming/Threading cover the cost of threading and context switching.  The actual cost depends on CPU and OS.